airflow/utils/context.py

```diff
@@ -161,54 +178,89 @@ class OutletEventAccessor:
     :meta private:
     """
 
-    raw_key: str | Asset | AssetAlias
+    key: str | BaseAssetUniqueKey
     extra: dict[str, Any] = attrs.Factory(dict)
     asset_alias_events: list[AssetAliasEvent] = attrs.field(factory=list)
 
-    def add(self, asset: Asset | str, extra: dict[str, Any] | None = None) -> None:
+    def add(self, asset: str | Asset, extra: dict[str, Any] | None = None) -> None:
         """Add an AssetEvent to an existing Asset."""
         if isinstance(asset, str):
-            asset_uri = asset
-        elif isinstance(asset, Asset):
-            asset_uri = asset.uri
-        else:
+            asset = Asset(asset)
+        elif not isinstance(asset, Asset):
             return
 
-        if isinstance(self.raw_key, str):
-            asset_alias_name = self.raw_key
-        elif isinstance(self.raw_key, AssetAlias):
-            asset_alias_name = self.raw_key.name
+        if isinstance(self.key, AssetAliasUniqueKey):
+            asset_alias_name = self.key.name
+        elif isinstance(self.key, str):
+            # TODO: deprecate string access
+            asset_alias_name = self.key
```

Comment:
We should remove this in 3.0 and deprecate in 2.11.

Reply:
Yep, https://github.com/apache/airflow/blob/99e713efa398bef9d76eb4e8145538e828a05720/airflow/utils/context.py#L227 has already deprecate it

Reply:
this has been removed as well